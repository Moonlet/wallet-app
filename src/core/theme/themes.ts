export enum Theme {
    dark = 'dark',
    light = 'light'
}
