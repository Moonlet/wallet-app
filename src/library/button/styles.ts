import { StyleSheet } from 'react-native';
import { ITheme } from '../../core/theme/itheme';

export default (theme: ITheme) =>
    StyleSheet.create({
        button: {
            paddingVertical: 10,
            borderWidth: 1,
            borderColor: theme.colors.accent,
            borderRadius: 6,
            paddingHorizontal: 6
            // flex: 1,
            // flexBasis: 0
        },

        text: {
            textAlign: 'center',
            color: theme.colors.accent
        },

        buttonPrimary: {
            backgroundColor: theme.colors.accent
        },

        textPrimary: {
            color: theme.colors.appBackground
        }
    });
