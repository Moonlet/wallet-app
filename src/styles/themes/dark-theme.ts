import { ITheme } from '../../core/theme/itheme';
import { COLORS } from '../colors';

export const darkTheme: ITheme = {
    dark: true,
    shadowGradient: ['#00000000', '#000000CC'],

    fontSize: {
        small: 12,
        regular: 16,
        large: 34
    },

    colors: {
        primary: COLORS.GRAY,
        accent: COLORS.AQUA,
        text: COLORS.WHITE,
        textDarker: COLORS.LIGHT_GRAY,
        positive: COLORS.HOT_GREEN,
        negative: COLORS.HOT_RED,

        cardBackground: COLORS.DARK_GRAY,
        appBackground: COLORS.COD_GRAY,
        settingsDivider: COLORS.BLACK
    }
};
