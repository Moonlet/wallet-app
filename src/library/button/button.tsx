import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import { Text } from '../index';

import stylesProvider from './styles';
import { withTheme } from '../../core/theme/with-theme';
import { Icon } from '../../components/icon';
import { normalize, BASE_DIMENSION } from '../../styles/dimensions';

/**
 * default:    text and border 'accent' color
 * primary:    filled button with 'accent' color bg
 * secondary:  text and border gray/white
 */

export interface IButtonProps {
    children?: any;
    wrapperStyle?: any;
    style?: any;
    primary?: boolean;
    secondary?: boolean;
    disabled?: boolean;
    disabledSecondary?: boolean;
    onPress?: any;
    styles: ReturnType<typeof stylesProvider>;
    leftIcon?: string;
    onPressIn?: any;
    onPressOut?: any;
    bottomLabel?: string;
}

export const ButtonComponent = (props: IButtonProps) => (
    <View style={[props.styles.container, props.wrapperStyle]}>
        <TouchableOpacity
            disabled={props.disabled || props.disabledSecondary}
            onPress={props.onPress}
            onPressIn={props.onPressIn}
            onPressOut={props.onPressOut}
            style={[
                props.styles.button,
                props.primary && props.styles.buttonPrimary,
                props.secondary && props.styles.buttonSecondary,
                props.disabled && props.styles.buttonDisabled,
                props.disabledSecondary && props.styles.buttonDisabledSecondary,
                props.style,
                {
                    paddingVertical: props?.leftIcon
                        ? BASE_DIMENSION
                        : BASE_DIMENSION + BASE_DIMENSION / 2
                }
            ]}
        >
            {props?.leftIcon && (
                <Icon
                    name={props.leftIcon}
                    size={normalize(32)}
                    style={[
                        props.styles.leftIcon,
                        props.primary && props.styles.leftIconPrimary,
                        {
                            marginRight: props?.children ? BASE_DIMENSION : 0,
                            paddingHorizontal: props?.bottomLabel ? BASE_DIMENSION : 0,
                            paddingVertical: props?.bottomLabel ? BASE_DIMENSION / 2 : 0
                        }
                    ]}
                />
            )}
            {props?.children && (
                <Text
                    style={[
                        props.styles.text,
                        props.primary && props.styles.textPrimary,
                        props.secondary && props.styles.textSecondary,
                        props.disabled && props.styles.textDisabled,
                        props.disabledSecondary && props.styles.textDisabledSecondary
                    ]}
                >
                    {props.children}
                </Text>
            )}
        </TouchableOpacity>
        {props?.bottomLabel && <Text style={props.styles.bottomLabel}>{props.bottomLabel}</Text>}
    </View>
);

export const Button = withTheme(stylesProvider)(ButtonComponent);
