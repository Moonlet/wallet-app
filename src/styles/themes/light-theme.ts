import { ITheme } from '../../core/theme/itheme';
import { darkTheme } from './dark-theme';

export const lightTheme: ITheme = darkTheme;
