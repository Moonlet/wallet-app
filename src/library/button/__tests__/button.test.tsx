import React from 'react';
import { ButtonComponent, IButtonProps } from '../button';
import { darkTheme } from '../../../styles/themes/dark-theme';
import styleProvider from '../styles';

import { shallow } from 'enzyme';

const props: IButtonProps = {
    styles: styleProvider(darkTheme)
};

describe('library button', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<ButtonComponent {...props}>dummy</ButtonComponent>);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
