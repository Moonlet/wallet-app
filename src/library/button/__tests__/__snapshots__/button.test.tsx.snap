// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`library button renders correctly 1`] = `
"<TouchableOpacity onPress={[undefined]} style={{...}} activeOpacity={0.2}>
  <Component style={{...}}>
    dummy
  </Component>
</TouchableOpacity>"
`;
